src/cmd/cli/command/compose.go

```diff
@@ -368,20 +410,38 @@ func makeComposeConfigCmd() *cobra.Command {
 		Short: "Reads a Compose file and shows the generated config",
 		RunE: func(cmd *cobra.Command, args []string) error {
 			loader := configureLoader(cmd)
-			project, err := loader.LoadProject(cmd.Context())
+			ctx := cmd.Context()
+			provider, err := getProvider(ctx, loader)
 			if err != nil {
 				return err
 			}
 
-			provider, err := getProvider(cmd.Context(), loader)
-			if err != nil {
-				return err
+			project, loadErr := loader.LoadProject(ctx)
+			if loadErr == nil {
+				_, _, err = cli.ComposeUp(ctx, project, client, provider, compose.UploadModeIgnore, defangv1.DeploymentMode_UNSPECIFIED_MODE)
+				if err != nil && errors.Is(err, cli.ErrDryRun) {
+					return nil
+				}
+
+				loadErr = err
 			}
 
-			if _, _, err := cli.ComposeUp(cmd.Context(), project, client, provider, compose.UploadModeIgnore, defangv1.DeploymentMode_UNSPECIFIED_MODE); !errors.Is(err, cli.ErrDryRun) {
-				return err
+			if loadErr != nil {
+				if !nonInteractive {
+					project, err = CreateDebugProject(loader)
+					if err != nil {
+						return err
+					}
+				}
+
+				track.Evt("Debug Prompted", P("reason", loadErr))
+				// Call the AI debug endpoint using the original command context (not the tailCtx which is canceled)
+				if nil == cli.InteractiveDebug(ctx, client, provider, "", project, nil, loadErr) {
+					return nil // don't show the defang hint if debugging was successful
+				}
 			}
-			return nil
+
+			return loadErr
```

Comment:
Isn't `loadErr` guaranteed to be `nil` here?
```suggestion
			return  nil
```

Reply:
if it's not interactive then we don't process the loadErr through the AI Debugger